### PR TITLE
Wording changes in man page

### PR DESCRIPTION
**Pull request by mppf**
Tidy up wording for man page relating to optimization.

### CODE REVIEW

**Comment by mppf** [comment]
@bradcray perhaps we could discuss the man page wording here? Generally throughout this document it talks about the C backend compiler -- but pretty much all of the options apply also to LLVM code generation as well. I'm not sure how to clear that up without making a lot of changes. Generally speaking though, the LLVM backend supports the same options and just modifies LLVM compilation in the same way that clang would (so there sortof is a C compiler the options go to...)

File: man/chpl.txt [hunk 1]
```diff
@@ -103,9 +103,7 @@ OPTIONS
   --[no-]dead-code-elimination   Enable [disable] dead code elimination.
 
   --fast            Turns off all runtime checks using --no-checks, turns
-                    on -O and --specialize, and enables most other compiler
-                    optimizations described in this section (--no-ieee-float
-                    and --cache-remote are notable exceptions).
+                    on -O, --specialize and --vectorize.
```
Comment (bradcray) [inline man/chpl.txt]: I'd forgotten about --vectorize.  Personally, I'd add the Oxford comma here.

File: man/chpl.txt [hunk 2]
```diff
@@ -114,7 +112,7 @@ OPTIONS
   --[no-]ieee-float   Disable [enable] optimizations that may affect IEEE
                     floating point conformance. The default is whatever level
                     of optimization/IEEE floating point support your C compiler
-                    provides with the C compiler optimization provided.
+                    provides at a given optimization level.
```
Comment (bradcray) [inline man/chpl.txt]: "provides at a given..." still seems vague (for me it begs the question "which one?").  I suggested "the given" as a means of suggesting that someone was specifying it without going to the trouble of saying who...  (the answer effectively being "typically 'chpl' as steered by the user's choice of other options").  We could be slightly more precise by saying "provides for the optimization level specified by 'chpl'".  

(To be fully precise, I think we'd need to refer to the -O option, the -cflags option, environment variables, etc. but that feels like overkill to me.)

**Comment by bradcray** [comment]
I hadn't looked in-depth enough to realize that there are already lots of references to the back-end C compiler, and agree that that's a bigger issue than we'd probably want to tackle here.  I've been trying to change my personal use of "C compiler" to "back-end compiler" in cases where it isn't specific to C, which is what made me suggest the change.

**Pull request merged**